src/tokenizer.ts

```diff
@@ -1485,12 +1480,45 @@ export class Tokenizer extends DiagnosticEmitter {
         ) {
           ++this.pos;
         }
-        while (this.pos < end && isDecimalDigit(text.charCodeAt(this.pos))) {
-          ++this.pos;
+        this.readDecimalFloatPartial();
+      }
+    }
+    let floatString = text.substring(start, this.pos).replace(/_/g, '');
```

Comment:
```suggestion
    let floatString = text.substring(start, this.pos).replaceAll("_", "");
```

Reply:
This seems to cause an error `TS2339: Property 'replaceAll' does not exist on type 'string'.`

Reply:
just ignore this via `@ts-ignore`

Reply:
btw it should be [exists in `esnext`](https://github.com/microsoft/TypeScript/pull/37476)

Reply:
So I suggest change to `esnext` instead `es2016` [here](https://github.com/AssemblyScript/assemblyscript/blob/master/src/tsconfig.json#L7) instead using `// @ts-ignore`

Reply:
`// @ts-ignore` worked, but `esnext` did not.